src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -322,7 +331,38 @@ class AppsController(
     }
   }
 
-  private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = ???
+  @SuppressWarnings(Array("all")) // async/await
+  private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = {
+    groupManager.app(appId) match {
+      case Some(app) =>
+        authorized(ViewRunSpec, app).apply {
+          val tasksF = async {
+            val instancesBySpec = await(instanceTracker.instancesBySpec)
+            await(runningTasks(Set(appId), instancesBySpec).runWith(Sink.seq)).map(_.toRaml)
+          }
+          onSuccess(tasksF) { tasks =>
+            complete(raml.EnrichedTasksList(tasks))
+          }
+        }
+      case None =>
+        reject(Rejections.EntityNotFound.noApp(appId))
+    }
+  }
+
+  @SuppressWarnings(Array("all")) // async/await
+  private def runningTasks(appIds: Set[PathId], instancesBySpec: InstancesBySpec): Source[EnrichedTask, NotUsed] = {
+    Source(appIds).mapAsync(1) { appId =>
```

Comment:
Hm, I don't see the benefit of creating a stream here. Why not simply use `Future.sequence`?

Reply:
The main benefit of having a stream instead of `Future.sequence` is that `sequence` will fire everything at once, and `mapAsync` will respect backpressure.

Reply:
Ah, ok cool!